src/test/java/com/google/sps/SentimentServletTest.java

```diff
@@ -53,97 +61,159 @@ public void setUp() throws IOException {
    * A Json String in the form of a multi-element array should be converted to a String of period
    * separated comments. A sentiment score should be calculated and the score should be printed in
    * a Json object. 
-   * Ex. "[foo, bar]" -> "{"commentScore":0.05}" 
    */
   @Test
-  public void jsonMultiElementArrayPrintsVideoAnalysisJsonObject() throws IOException {
-    String json = "[\"a\",\"b\",\"c\"]"; // "["a","b","c"]"
-    when(requestMock.getReader()).thenReturn(new BufferedReader(new StringReader(json)));
+  public void printsVideoAnalysisJsonObject() throws IOException {
+    List<String> comments = Arrays.asList("foo","bar","foobar");
+    String captions = "foo. bar. foobar.";
+    
+    when(requestMock.getReader()).thenReturn(new BufferedReader(new StringReader(VIDEO_ID)));
+    when(commentServiceMock.getCommentsFromId(VIDEO_ID)).thenReturn(comments);
+    when(captionServiceMock.getCaptionFromId(VIDEO_ID)).thenReturn(captions);
     when(languageServiceMock.analyzeSentiment(any(Document.class)).getDocumentSentiment())
         .thenReturn(sentimentMock);
     when(sentimentMock.getScore()).thenReturn(SCORE);
     doNothing().when(languageServiceMock).close();
 
+    sentimentServlet.doPost(requestMock, responseSpy);
+
     sentimentServlet.doGet(requestMock, responseSpy);
 
-    Assert.assertEquals(stringWriter.toString(), "{\"commentScore\":0.05}\n");
+    Assert.assertEquals(stringWriter.toString(), "{\"score\":0.05}\n");
```

Comment:
Could you build the expected string inside the unit test using the SCORE variable. Its not obvious that 0.05 is the value of SCORE by just reading the unit test.